src/libopensc/reader-pcsc.c

```diff
@@ -870,31 +875,32 @@ part10_find_property_by_tag(unsigned char buffer[], int length,
  */
 static size_t part10_detect_max_data(sc_reader_t *reader, SCARDHANDLE card_handle)
 {
-    u8 rbuf[256];
-    DWORD rcount = sizeof rbuf;
-    struct pcsc_private_data *priv;
-	/* 0 means no limitations */
-    size_t max_data = 0;
+	u8 rbuf[256];
+	DWORD rcount = sizeof rbuf;
+	struct pcsc_private_data *priv = NULL;
+	/* 0 means extended APDU not supported */
+	size_t max_data = 0;
 	int r;
 
-    if (!reader)
-        goto err;
-    priv = GET_PRIV_DATA(reader);
-    if (!priv)
-        goto err;
+	if (!reader)
+		goto err;
+	priv = GET_PRIV_DATA(reader);
+	if (!priv)
+		goto err;
 
-    if (priv->get_tlv_properties && priv->gpriv) {
+	if (priv->get_tlv_properties && priv->gpriv) {
 		if (SCARD_S_SUCCESS != priv->gpriv->SCardControl(card_handle,
-					priv->get_tlv_properties, NULL, 0, rbuf, sizeof(rbuf),
-					&rcount)) {
-			sc_debug(reader->ctx, SC_LOG_DEBUG_NORMAL,
-				   	"PC/SC v2 part 10: Get TLV properties failed!");
+				priv->get_tlv_properties, NULL, 0, rbuf, sizeof(rbuf), &rcount)) {
+			sc_log(reader->ctx, "PC/SC v2 part 10: Get TLV properties failed!");
 			goto err;
 		}
 
 		r = part10_find_property_by_tag(rbuf, rcount,
 				PCSCv2_PART10_PROPERTY_dwMaxAPDUDataSize);
-		if (r >= 0)
+		sc_log(reader->ctx, "get dwMaxAPDUDataSize property returned %i", r);
+
+		/* 256 < X <= 0x10000: short and extended APDU of up to X bytes of data */
+		if (r > 0x100 && r <= 0x10000)
```

Comment:
There are readers that only support up to 251 bytes of data, which would not be recognized here. This should check for `r > 0` instead of `r > 0x100`.